### ISSUE TITLE
Improve voting

### ISSUE THREAD

**Issue by NathanYocum** [LectureGoggles/FlaskAPI#41]
* Vote choice needs to only be `1, 0, or -1`. 1 is a new upvote, -1 is a new downvote, 0 is removing the current vote
* We need to make it so users can't constantly call `vote/onPost` to upvote a post. If a user has already upvoted, they are only allowed to use `vote_choice` of 0 or -1, if they have already downvoted, they can only use 1 or 0.

**Comment by NathanYocum** [comment]
Current behavior: If I do 
```javascript
axios.post(`${urlToUse}/v1/vote/onPost/${id}/`, { vote_choice: 5 }, ...
```
every time I send an upvote it does +5, for example. This is something that should be verified on API side because it'd be impossible to validate efficiently on the frontend.

I can also spam the upvote button and it constantly increases the vote count

**Comment by zackjohnson8** [comment]
Good call. When I was looking at it yesterday it just seemed like spaghetti code. I'll get to this soon and shouldn't take long to fix.

**Comment by NathanYocum** [comment]
Another important thing is that I need a good way to keep persistent state of upvotes. Will probably need to be something returned from `/v1/post/get...`

**Comment by NathanYocum** [comment]
Closed by #43 #44 #46 #47 and #48 :grimacing:
